code/modules/shuttle/supply.dm

```diff
@@ -219,6 +216,9 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 			message_admins("\A [spawning_order.pack.name] ordered by [ADMIN_LOOKUPFLW(spawning_order.orderer_ckey)], paid by [from_whom] has shipped.")
 		purchases++
 
+		SSshuttle.shopping_list -= spawning_order
+		qdel(spawning_order)
```

Comment:
```suggestion
		if(!spawning_order.pack.goody)
			SSshuttle.shopping_list -= spawning_order
			qdel(spawning_order)
```
Goody orders are stored in a `goodies_by_buyer` list, which is looped through near the end of this proc. They've to be deleted after that, not before.

Reply:
You right. Then it is best to move these 2 lines after looping over goody orders list else if the order is a goody it won't get deleted